src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -35,7 +35,7 @@ internal struct Builder
             /// <summary>
             /// True if the fields of the builder are unset.
             /// </summary>
-            internal bool IsDefault => _defaultType is null && _nullableAnnotation == 0 && (_extensions == null || _extensions == Extensions.Default);
+            internal bool IsDefault => _extensions == null;
```

Comment:
This needs to be volatile.Read. And I don't understand why we were previously allowing Extensions.Default.

Reply:
This is one case where I don't think we strictly need `Volatile.Read`. I'm not opposed to adding it but having this be a flaky ready just means sometimes we go into `InterlockedInitialize` when we don't strictly need to. That code already has to be hardened against that case hence it's not a safety issue.

Reply:
I suppose that's true.